src/main/java/com/rsmart/kuali/kfs/cr/dataaccess/CheckReconciliationDao.java

```diff
@@ -73,4 +73,6 @@
      * @return
      */
     public List<PaymentGroup> getAllPaymentGroupForSearchCriteria(KualiInteger disbNbr, Collection<String> bankCodes);
+
+    public CheckReconciliation findByCheckNumber(String checkNumber);
```

Comment:
Is the check number enough of a constraint for uniqueness when searching for CR rows? If we don't expect the bank account number to change, then I think we will be okay.

Reply:
I'm not sure, would you recommend adding the bank code?

Reply:
I think adding bank code will be enough, unless the CR rows are using several different bank account numbers for the same bank (which I do not think is the case).